ts-client/src/amm/utils.ts

```diff
@@ -0,0 +1,469 @@
+import {
+  getAmountByShare,
+  calculateWithdrawableAmount,
+  VaultState,
+  getUnmintAmount,
+} from '@mercurial-finance/vault-sdk';
+import { BN, EventParser } from '@project-serum/anchor';
+import { ASSOCIATED_TOKEN_PROGRAM_ID, Token, TOKEN_PROGRAM_ID } from '@solana/spl-token';
+import {
+  AccountInfo,
+  Connection,
+  ParsedAccountData,
+  PublicKey,
+  SystemProgram,
+  SYSVAR_CLOCK_PUBKEY,
+  TransactionInstruction,
+} from '@solana/web3.js';
+import invariant from 'invariant';
+import { CURVE_TYPE_ACCOUNTS, ERROR, VIRTUAL_PRICE_PRECISION, WRAPPED_SOL_MINT } from './constants';
+import { ConstantProductSwap, StableSwap, SwapCurve, TradeDirection } from './curve';
+import {
+  AccountsInfo,
+  ApyState,
+  ParsedClockState,
+  PoolInformation,
+  PoolState,
+  SwapQuoteParam,
+  VirtualPrice,
+} from './types';
+
+export const getMaxAmountWithSlippage = (amount: BN, slippageRate: number) => {
```

Comment:
Can you add a comment as it supported only max to 2 decimal place. Eg: x.xx% slippage rate